internal/pkg/agent/application/actions/handlers/handler_action_policy_change.go

```diff
@@ -104,10 +109,56 @@ func (h *PolicyChangeHandler) Handle(ctx context.Context, a fleetapi.Action, ack
 		return errors.New(err, "could not parse the configuration from the policy", errors.TypeConfig)
 	}
 
+	var configNeedToBeSaved bool
+
 	h.log.Debugf("handlerPolicyChange: emit configuration for action %+v", a)
-	err = h.handleFleetServerHosts(ctx, c)
+	fleetClientCfg, newFleetClient, err := h.handleFleetServerHosts(ctx, c)
```

Comment:
Note to self: Instead of returning the new client, return a simple `func()` callback that can be called after the configuration has been correctly saved. Same thing can be done for handleLogLevel